File: 1395-minimum-time-visiting-all-points/minimum-time-visiting-all-points.cpp
class Solution {
public:

    int findDistance(const vector<int> &from,const vector<int> &to) 
    {
        int xDiff = abs(from[0] - to[0]);
        int yDiff = abs(from[1] - to[1]);

        return max(xDiff, yDiff);
    }

    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        int totalTime = 0;

        for(int i = 1; i < points.size(); i++) {
            totalTime += findDistance(points[i-1], points[i]);
        }

        return totalTime;
    }
};

File: 1395-minimum-time-visiting-all-points/minimum-time-visiting-all-points.py
class Solution:
    def findMinPoints(self, x1 : int, x2 : int, y1 : int , y2 : int) -> int:
        return max(abs(x2-x1), abs(y2-y1))

    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        res = 0
        x1, y1 = points.pop()

        while points:
            x2, y2 = points.pop()
            res += self.findMinPoints(x1,x2,y1,y2) 
            x1, y1 = x2, y2

        return res